test/TestHelpers/TaskUtils.cs

```diff
@@ -0,0 +1,29 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.EventFlow.TestHelpers
+{
+    public static class TaskUtils
+    {
+        private static readonly TimeSpan JustABit = TimeSpan.FromMilliseconds(20);
```

Comment:
Thanks for pointing out this method to me, I did not realize we had it.

But no, I do not think this is a perfect replacement, primarily because the SpinWait.SpinUntil() does not use Task Parallel Lib at all. With my custom thread scheduler used by the tests I want the scheduler to have a chance to do "something else" while we are waiting on condition.